File: src/ERP/acme.atena.domain/DTO/Product/Produto.cs
using acme.atena.domain.DTO.Inventory;
using acme.atena.domain.DTO.Product.Price;
using acme.atena.domain.DTO.Product.Sequence;
using System;
using System.Collections.Generic;
using System.Text;

namespace acme.atena.domain.DTO.Product
{
    public class Produto : AbstractEntity
    {
        public Produto()
        {
            VendasProdutos = new HashSet<VendaProduto>();
            ComprasProdutos = new HashSet<CompraProduto>();
            ValorProdutos = new HashSet<ValorProduto>();
            EntradaProdutoEstoques = new HashSet<EntradaProdutoEstoque>();
            SaidaProdutoEstoques = new HashSet<SaidaProdutoEstoque>();
            EstoqueProdutos = new HashSet<EstoqueProduto>();
        }

        public string Nome { get; set; }
        public string Descricao { get; set; }
        public long Codigo { get; set; }
        public DateTime? Validade { get; set; }
        public Guid TipoProdutoId { get; set; }
        public TipoProduto TipoProduto { get; set; }

        public virtual CodigoProduto CodigoProduto { get;set;}
        public virtual ICollection<ValorProduto> ValorProdutos { get; set; }
        public virtual ICollection<VendaProduto> VendasProdutos { get; set; }
        public virtual ICollection<CompraProduto> ComprasProdutos { get; set; }
        public virtual ICollection<EntradaProdutoEstoque> EntradaProdutoEstoques { get; set; }
        public virtual ICollection<SaidaProdutoEstoque> SaidaProdutoEstoques { get; set; }
        public virtual ICollection<EstoqueProduto> EstoqueProdutos { get; set; }
        public virtual ICollection<DevolucaoCompra> DevolucaoCompras { get; set; } = new HashSet<DevolucaoCompra>();
        public virtual ICollection<DevolucaoVenda> DevolucoesVenda { get; set; } = new HashSet<DevolucaoVenda>();

        public bool Atualizar(Produto produto)
        {
            bool temProdutoAtualizar = false;
            if (Nome != produto.Nome)
            {
                Nome = produto.Nome;
                temProdutoAtualizar = true;
            }
            if(Validade != produto.Validade)
            {
                Validade = produto.Validade;
                temProdutoAtualizar = true;
            }
            if(TipoProdutoId != produto.TipoProdutoId)
            {
                TipoProdutoId = produto.TipoProdutoId;
                temProdutoAtualizar = true;
            }
            if(Descricao != produto.Descricao)
            {
                Descricao = produto.Descricao;
                temProdutoAtualizar = true;
            }
            return temProdutoAtualizar;
        }
    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/product/produto.ts
import { EntradaProdutoEstoque } from './../inventory/entrada-produto-estoque';
import { SaidaProdutoEstoque } from './../inventory/saida-produto-estoque';
import { EstoqueProduto } from './../inventory/estoque-produto';
import { VendaProduto } from './venda-produto';
import { TipoProduto } from './tipo-produto';
import { DevolucaoCompra } from './devolucao-compra';
import { AbstractEntity } from "../abstract-entity";
import { CompraProduto } from "./compra-produto";
import { ValorProduto } from './price/valor-produto';
import { DevolucaoVenda } from './devolucao-venda';

export class Produto extends AbstractEntity{
  public nome:string;
  public descricao: string;
  public codigo: number;
  public validade: Date;
  public tipoProdutoId: string;
  public tipoProduto: TipoProduto;

  public  ValorProdutos:ValorProduto[];
  public  VendasProdutos: VendaProduto[];
  public  ComprasProdutos:CompraProduto[];
  public  EntradaProdutoEstoques: EntradaProdutoEstoque[];
  public  SaidaProdutoEstoques: SaidaProdutoEstoque[];
  public  EstoqueProdutos:EstoqueProduto[];
  public  DevolucaoCompras : DevolucaoCompra[];
  public  DevolucoesVenda :DevolucaoVenda[];
}
